### ISSUE TITLE
Overwatch League skins

### ISSUE THREAD

**Issue by Gonzom** [Js41637/Overwatch-Item-Tracker#188]
Noticed you've added the sprays/icons but have not yet added the skins.
Could you create "team" pages (like holiday events) to group all the skins of a specific team together in addition to adding the skins to each hero's page?

**Comment by Js41637** [comment]
Maybe one day.

**Comment by djcorny** [comment]
Hey..
If it's about the screenshots, then I can help.

I took screenshots from all hero skins (w/ same default pose), cropped them to 900x900px, named them like the other files in Item Tracker and uploaded them [here to dropbox](https://www.dropbox.com/sh/ipcu95xbg3rq6kb/AACZNEoB6hwJ3pFOPcG1Ihtta?dl=0). They are sorted by heroes, not by teams.

Feel free to use them

**Comment by inimigor** [comment]
So... Is this a work in progress or not even on the plate at the moment? Can I help with something?

**Comment by Js41637** [comment]
I just need to create a whole new section for them and ideally have 2 modes, group by team and group by hero because I won't be putting them on the hero pages as it'll clutter it up too much.

**Comment by Gonzom** [comment]
I think its a mistake not putting the skins in the hero page. When someone browsers a specific hero and wants to see all the skins, that's where he'll want to look, not ping-pong between two different screens.

**Comment by Js41637** [comment]
Well the game doesn't do it unless you own them and I'm not keen on adding another 12 skins to every hero. Could add an owl 'section' to the hero pages but then one of the 3 rows on hero pages will be super long or longer and look stupid. I'll add a button somewhere on the page to view OWL skins for a hero.

**Comment by inimigor** [comment]
I think adding it to the side bar would be a good option, or just in the "Hero Gallery" of the home page.

Adding a bar/button on the lower part of the page simply called "Overwatch League Skins".

I think it's a good compromise between not making it cluttered, and since it's not an event thing, I feel like it shouldn't take as priority of space as events either.

**Comment by crumins** [comment]
Please don't add League skins to the main functionality of this amazing tool. 99% of people will only ever have 1-5 skins unlocked... it'll only serve to clutter this tool that helps me know exactly what I'm missing for each character.

**Comment by Js41637** [comment]
Yeah this is why I am hesitant to do it as well. They'll definitely be on a seperate page where you can view OWL skins for each hero or each team (like it is ingame) and will have a button somewhere on the hero details page to view OWL skins

**Comment by Gonzom** [comment]
One more thing to consider though, if you don't add it, the values in the collection will not match the in-game one as it counts collected league skins.

Regarding "clutter" (that's an opinion, not a fact), you could add a toggle option of displaying these and let each person decide if he wants this visible or not.

**Comment by Js41637** [comment]
It only counts OWL skins if you have that skin unlocked, I could do the same, very painfully.

**Comment by Gonzom** [comment]
That's just not true crumins. I got 10 skins for Reinhardt, one of them is the League.

**Comment by Gonzom** [comment]
@Js41637 How hard/time investment would it be to create a "OWL Skins" category and have the option to show it or not via toggle/settings?

**Comment by Js41637** [comment]
It gets a lot more complicated when calculating item totals and such, the current method is already ridiculously overly complicated, it could be done though.

**Comment by crumins** [comment]
Don't do it.

**Comment by essidus** [comment]
I agree. If you're putting them in, make them live in their own space. I feel like it's really not something anyone will ever want a 100% collection of anyway.

**Comment by inimigor** [comment]
On the topic of the numbers added to the total only as you unlock something... 

 I think a separate function on the code that adds the total/owned values, and running a sum for that category when an item is checked/unchecked, and then adding the categories sum would be a great solution.  

This would be a clean solution, agnostic to category, so Less code, easily hero/event adjustable, and independent to the page you are at the moment of checking/unchecking.  Also, it's relatively quick to process when filtered in most ways (instead of checking, for example, 500 items sequentially when doing this, you check two sets of smaller much smaller number of items), by having the sum of the items by category first, and then a total sum of the tables, so just checking the total of "owned items" and "all items counted as part of the total" per table and adding them up for the total for each hero, since the tables are processed individually, and only when changes happen, this is usually faster to process. 

It's harder to implement initially, but scales a lot better as time goes on. And I'm sure there's a way to improve on the logic before applying, but I'd like to at least check first if you tried to do it this way, and if not... are you interested in trying? 


I know doing tables as separate objects would require working the backlog of pretty much everything, but I'm at the hospital with a friend, he's OK, but needs observation and company until he's out, and having "lots of free time"  but not so much to do is not that great either... 

I'd be able to check and fill the backlog of items these functions apply to, my notebook won't run Overwatch so I can't even farm stuff from the Lunar New Year event while here.

If needed, I'm used to programming Java, just a bit rusty.  


And, again, I agree with everyone not wanting them on the default hero page, too much stuff. 
But not putting it at all would suck even more! A separate page seems to be more than enough for this. 

 Even with a single skin, having the numbers not matching drove me insane, I unchecked one of the achievement sprays for the heroes I got OWL skins for, but it still haunts me at night.

This is a great tool and I'd be glad to help improve it!

**Comment by MrMaquette** [comment]
Would it be possible (down the road) to add a dropdown menu of the OWL skins for that hero at the bottom of the skin list and only selected ones are added to the main list? I'm sure it's a lot of work but thought I would just add my idea.

**Comment by catweasels** [comment]
To add my two cents, regarding OWL skins vs hero unlock count....
Would it be possible to just add a function to +1 the count? Like a box that raises your personal unlock for that hero by +#
Also, i understand not adding the team skins, but what about the twitch unlockable skins, the non-team owl ones?

**Comment by GlassedSilver** [comment]
Folks should learn to speak for themselves... "I don't think anyone ever...", "It'll only clutter", etc...

Honestly, if you want to go about this in a clean manner, just have the user put them in a special section and all that they have appear in the hero list... you know... Just like in the game.

Those who only get a couple don't have to be bothered by long lists I guess and those who go ham on a single hero know what they are getting into.

**Comment by Gonzom** [comment]
Is there any update on this?

**Comment by Js41637** [comment]
There's been a bit of discussion recently as well as a PR submitted by someone, I was going to look into adding it however there has been some concerning developments. A lot of World Cup data has been detected in the game, i am unsure if they are planning on making these skins available for purchase, I certainly hope not because it will make things a bit harder not to mention increase the skin count substantially. I am leaning towards on waiting until some more information is available. Next qualifiers stage of World Cup is in a few weeks and the official World Cup is on at Blizzcon.

**Comment by Js41637** [comment]
Oh right and there has been talk that some teams might be changing their color which will really complicate things. I don't think it's possible for them to just change the color of a team as people payed money for the original team colors, forcing a change seems very anti-consumer and has a potential for a lot of backlash which means if teams do end up changing colors, it means there will be another "set" of skins for the team with the old ones remaining. I'm not sure if this will happen it will require a lot of changes to the current plan for implementation.

**Comment by Gonzom** [comment]
I'm not sure I see how the changing of colors complicates this. I have no idea what your current plan of implementation is/was.

1. Say you wanted to set it up by team sections, then you just have: "Team A", "Team B", "Team C", "Team D (2018)", "Team D (2019".
2. Say you wanted to set it by hero, then it doesn't matter how many skins are added.

Regarding the off-topic in-game issue, I think Blizz's best option would be to give a refund option to anyone who's skin changed colors. Easy to handle and has no issues in-game. If the team gets % of money from those skins, then they just "owe" that amount back to Blizzard.

**Comment by DracoClaw** [comment]
Now that 8 new teams are joining the league, does that change the way adding these would work?

**Comment by Js41637** [comment]
Nah, someone already confirmed that no teams will be getting new colors which is good. I'm more scared about World Cup teams getting skins, there's already some data in the game to suggest they could be getting skins but it could also just be for the World Cup.

**Comment by DracoClaw** [comment]
So it's been a while, any updates on this?
